### PR TITLE
Make MedianTask number of allowed input errors configurable

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/services/pipeline/graph.go [hunk 1]
```diff
@@ -65,6 +65,11 @@ func (g TaskDAG) TasksInDependencyOrder() ([]Task, error) {
 			return nil, err
 		}
 
+		err = task.ApplyDefaults(node.attrs, g, *node)
```
Comment (connorwstein) [inline core/services/pipeline/graph.go:68]: Isn't this more of a "SetAttributes" vs ApplyDefaults naming wise ?

File: core/services/pipeline/task.bridge.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ func (t *BridgeTask) Type() TaskType {
 	return TaskTypeBridge
 }
 
+func (t *BridgeTask) ApplyDefaults(inputValues map[string]string, g TaskDAG, self taskDAGNode) error {
```
Comment (connorwstein) [inline core/services/pipeline/task.bridge.go:31]: If we define this return nil on the BaseTask we don't have to define it on every task, just override where needed

File: core/services/pipeline/task.median.go [hunk 3]
```diff
@@ -21,6 +22,16 @@ func (t *MedianTask) Type() TaskType {
...
 }
 
+func (t *MedianTask) ApplyDefaults(inputValues map[string]string, g TaskDAG, self taskDAGNode) error {
+	if _, exists := inputValues["allowedFaults"]; !exists {
+		if len(self.inputs()) == 0 {
+			return errors.Wrapf(ErrWrongInputCardinality, "MedianTask requires at least 1 input")
+		}
+		t.AllowedFaults = uint64(len(self.inputs()) - 1)
+	}
+	return nil
```
Comment (connorwstein) [inline core/services/pipeline/task.median.go:32]: so if its not set, then we tolerate 0 faults instead of 50% like before?

Comment (spooktheducks) [reply core/services/pipeline/task.median.go:32]: No, if it's not set, we tolerate `n - 1` faults where `n` is the number of tasks feeding into the medianizer.

File: core/services/pipeline/task.bridge.go (hunk 2 above)
Comment (connorwstein) [reply core/services/pipeline/task.bridge.go:31]: spoke offline - want to make it explicit

File: core/services/pipeline/graph.go (hunk 1 above)
Comment (connorwstein) [inline core/services/pipeline/graph.go:68]: nit: could move this above the UnmarshalTaskFromMap avoid the "set default if not set" and just have it naturally override the default assuming that mapstructure overrides

**Review by samsondav — DISMISSED** [review]

**Review by connorwstein — DISMISSED** [review]

**Review by connorwstein — APPROVED** [review]

**Pull request merged**